### DISCUSSION TITLE
Cross Validate 2 Inputs force same value error messages

### DISCUSSION THREAD

**Discussion by simann** [react-hook-form/react-hook-form#10206] [Q&A] [answered]
I am new to react-hook-form and currently i am having some trouble.

It is a pretty common use case and im sure somehow has the answer. I just cant figure it out.

Case:
2 Input Fields A & B, both are required and its required that they have the same value.

Input A contains "a"
Input B cointains "b"
onSubmit i will get the expected error "fields dont match" on both inputs.

Now i change the value of B to "a" and the error message on B disappears but its still showing on A.

I suspect thats because B got changed, thus B is revalidated but A is not. I found `trigger` in the documentation but im not sure where i should call it or if it even helps.

What i would like to achieve is:
Change the value of B to the value of A and both error messages should disappear.

Sandbox:
https://codesandbox.io/s/pedantic-colden-84c1zk?file=/src/App.js

**Comment by devkayes** [comment, marked as answer]
To solve this issue you need to watch field values with `watch()` .  And make some condition if not match set Custom error with `setError()`, if match then clear the error with `clearErrors()`. 

I thing that's the code you want 👍
```
import "./styles.css";
import { useForm } from "react-hook-form";
import { useEffect } from "react";

export default function App() {
  const {
    register,
    handleSubmit,
    getValues,
    formState: { errors },
    watch,
    setError,
    clearErrors
  } = useForm({});

  const firstField = watch("first");
  const secondField = watch("second");

  useEffect(() => {
    if (firstField === secondField) {
      clearErrors();
      console.log("Enter");
      return;
    } else {
      setError(
        "customError",
        { message: "don't match" },
        { shouldFocus: true }
      );
    }
  }, [secondField, firstField, clearErrors, setError]);

  const submit = (data) => {
    console.log(data);
  };

  return (
    <form onSubmit={handleSubmit(submit)}>
      <label>Values same values</label>

      <input
        style={{
          display: "block"
        }}
        {...register("first", {
          required: true,
          validate: () => {
            return getValues("first") === getValues("second");
          }
        })}
      ></input>

      <input
        style={{
          display: "block"
        }}
        {...register("second", {
          required: true,
          validate: () => {
            return getValues("first") === getValues("second");
          }
        })}
      ></input>

      <p> isMatch: {errors.customError?.message || "Match"} </p>

      <input
        style={{
          display: "block"
        }}
        type="submit"
      ></input>
    </form>
  );
}
```

**Reply by simann** [reply]
Thanks for the example!

This works for me.
